### ISSUE TITLE
use hyperdrive-stats

### ISSUE THREAD

**Issue by juliangruber** [datproject/dat#541]
For more reliable stats: https://github.com/juliangruber/hyperdrive-stats

**Comment by joehand** [comment]
@juliangruber sorry, didn't see you'd self-assigned this. I had started this last week but had a few edge cases to work out before I pushed it.

I opened a [PR on dat-js](https://github.com/joehand/dat-js/pull/14). How does that look to you? I'm not really using the capabilities of hyperdrive-index there, but was seeing some weird behavior around partial downloads if I didn't start at the beginning of the index each time.

On the CLI side, we need to change download progress to use blocks (instead of bytes). I had started that awhile ago (on the [fix/blocks-progress](https://github.com/datproject/dat/commit/408b1b6bc63ee440632a72e78e5bb1ddbff58f68) branch). The display will have to change a bit since we can't show bytes progress reliably. That branch works with my dat-js PR. Or let me know what you were thinking for the CLI.
